astroid/nodes/scoped_nodes/scoped_nodes.py

```diff
@@ -2138,6 +2138,9 @@ def my_meth(self, arg):
     _other_other_fields = ("locals", "_newstyle")
     _newstyle = None
 
+    is_dataclass = False
+    """Whether this class is a dataclass."""
```

Comment:
This should be an instance attribute, i.e. set in `__init__`.
Could you add a type annotation, too?

Furthermore, I believe the attribute name should also be added to `_other_fields`.